R/if_not_else_linter.R

```diff
@@ -0,0 +1,108 @@
+#' Block statements like if (!A) x else y
+#'
+#' `if (!A) x else y` is the same as `if (A) y else x`, but the latter is
+#'   easier to reason about in the `else` case. The former requires
+#'   double negation that can be avoided by switching the statement order.
+#'
+#' This only applies in the simple `if/else` case. Statements like
+#'   `if (!A) x else if (B) y else z` don't always have a simpler or
+#'   more readable form.
+#'
+#' It also applies to [ifelse()] and the package equivalents
+#'   `dplyr::if_else()` and `data.table::fifelse()`.
+#'
+#' @param exceptions Character vector of calls to exclude from linting.
+#'   By default, [is.null()], [is.na()], and [missing()] are excluded
+#'   given the common idiom `!is.na(x)` as "x is present".
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "if (!A) x else y",
+#'   linters = if_not_else_linter()
+#' )
+#'
+#' lint(
+#'   text = "ifelse(!is_treatment, x, y)",
+#'   linters = if_not_else_linter()
+#' )
+#'
+#' lint(
+#'   text = "if (!is.null(x)) x else 2",
+#'   linters = if_not_else_linter(exceptions = character())
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "if (A) x else y",
+#'   linters = if_not_else_linter()
+#' )
+#'
+#' lint(
+#'   text = "ifelse(is_treatment, y, x)",
+#'   linters = if_not_else_linter()
+#' )
+#'
+#' lint(
+#'   text = "if (!is.null(x)) x else 2",
```

Comment:
I think this example still needs to be changed?

Reply:
This example is meant to show the default behavior that `!is.null()` passes without linting